### ISSUE TITLE
FOX-ETH LP: Implement DeFi opportunity

### ISSUE THREAD

**Issue by gomesalexandre** [shapeshift/web#2270]
### Overview

**Note: This depends on https://github.com/shapeshift/web/issues/2259 and ideally would be worked on after, but it is possible to work on both in parallel at assignee's discretion.**

This feature to provide liquidity for FOX token and Ethereum currently exists on [fox.shapeshift.com/fox-farming](https://fox.shapeshift.com/fox-farming)

The goal of this bounty is to build that feature into our [app.shapeshift.com](http://app.shapeshift.com/) interface, into the Defi section: https://app.shapeshift.com/#/defi

### References and additional details

- The new DeFi UI should be used to implement this https://github.com/shapeshift/web/pull/2116
- The shoehorned helpers from the foxfarm repo (https://github.com/shapeshift/web/issues/2259) will be used to interact with the smart contracts

- A new provider will most likely need to be created under `src/features/defi/providers` with routes similar to the YearnPath routes or the FoxyPath ones. You’ll need to use a new `earnType`, as currently, the only two types of defi earn types we have is token_staking for FOXy, and vault for yearn so this could be e.g liquidity_providing
- Copying the whole src/features/defi/providers/yearn and renaming variables to match the new earn type / starting to remove/rename stuff is one way to go with it, rather than actually rewriting it.

- Note that we have ReusableXYZ components for DeFi that should be use as building blocks for making DeFi modal flows

Most of the presentational stuff should be handled by the DeFi UI abstraction already, and the missing pieces can be added by composing it. The flow should be the same one we have for DeFi vaults and most of the logic will come down to actually interacting with the LP pools. Interacting with UniV2 LPs as done through https://fox.shapeshift.com/fox-farming (https://github.com/shapeshift/foxfarm) but historically, ShapeShift has also started to integrate in web with the FoxPage: https://app.shapeshift.com/#/fox

Note: Despite what the path of these two modules suggests, making an opportunity available through Overview/Earn tabs is done through the <EarnOpportunities /> and <AllEarnOpportunities /> components under src/components/StakingVaults - these are not just related to the vault type but really to any opportunity

### Acceptance Criteria

- FOX/ETH liquidity providing opportunity is added to Earn tab in the DeFi section as well as FOX and ETH asset pages
(includes metadata of APR, TVL, fiat value of connected wallet's balance, and type: Liquidity Pool)
- Users can deposit FOX & ETH into the FOX/ETH Uniswap v2 liquidity pool (UI/UX matches mockup)
- The feature is hidden under a feature flag: https://github.com/shapeshift/web/issues/2269

### Feature Spec

[https://www.notion.so/shapeshift/FOX-ETH-LP-on-app-shapeshift-com-329b07d4d42a4093876632887749456d](https://www.notion.so/FOX-ETH-LP-on-app-shapeshift-com-329b07d4d42a4093876632887749456d)

### UX

https://www.figma.com/proto/D2DZI92GLfpCGEwLOmU6ps/OSMOSIS

The flow will be almost identical to the one in the aforementioned Figma mockups, except if the user has not already done so, there will need to be an approval step.

Here is what the approval step looks like for reference:

![image](https://user-images.githubusercontent.com/17035424/181635961-ad1119a4-74fe-4dd7-bfd6-b2c5024b2cf0.png)

### Estimated effort

40 hours
